build_tools/aws-sdk-code-generator/lib/aws-sdk-code-generator/waiter.rb

```diff
@@ -90,6 +96,32 @@ def acceptors(acceptors)
         ).format(acceptors: acceptors)
       end
 
+      # Underscore and symbolize all keys in hash
+      def convert_keys(hash)
+        hash.each_with_object({}) do |(key, value), result|
+          converted_key = convert_key(key)
+          result[converted_key] = convert_value(value)
+        end
+      end
+
+      def convert_key(key)
+        key.is_a?(String) ? Underscore.underscore(key).to_sym : key
+      end
+
+      def convert_value(value)
+        case value
+        when Hash  then convert_keys(value)
+        when Array then convert_array(value)
+        else value
+        end
+      end
+
+      # Convert keys for every object in array
+      def convert_array(array)
+        array.map do |elem|
+          elem.is_a?(Hash) ? convert_keys(elem) : elem
```

Comment:
You can just call convert_value here right? I would do array.collect { |e| convert_value(e) }